### ISSUE TITLE
Multiple instances on Synology NAS

### ISSUE THREAD

**Issue by DoctorDri** [rvolz/BicBucStriim#268]
Hi,

I got BicBucStriim up and running on a Synology 918+ server. Installed Apache 2.2 and PHP 5.6, configured web station, and everything works fine.

But then I tried to have a second instance of BBS, because I have a Calibre library for English books and a second one for books in Spanish. So I created a folder web/bbs2, copied last version of BBS into it... but nothing happens, I just get the error: "This page isn’t working. 192.168.1.40 is currently unable to handle this request. HTTP ERROR 500"

Any ideas? The first BBS works fine.

**Comment by rvolz** [comment]
You should make sure, that the new directory `web/bbs2` has the same access rights and owner as the the original. If that is the case and the the error is still there you could temporarily [enable debug mode](https://github.com/rvolz/BicBucStriim/wiki/Debug-mode) to get more error information.

**Comment by DoctorDri** [comment]
I checked, same permissions as the original folder. So I enabled debug mode. Nothing happens, same error.

**Comment by rvolz** [comment]
The error info might be in the log files, which are located in the `data` directory.

**Comment by DoctorDri** [comment]
What kind of file should I be looking for?

**Comment by rvolz** [comment]
There should be log files, with names like YYYY-MM-DD.log.

**Comment by DoctorDri** [comment]
There's none, and I think I edited web/bbs2/index.php correctly:
![capture](https://user-images.githubusercontent.com/21368366/34715535-4915d4b6-f52d-11e7-9f1a-404eb5d4b69d.JPG)

**Comment by DoctorDri** [comment]
I noticed that the Package Center on the NAS shows a different version of BBS:

![capture2](https://user-images.githubusercontent.com/21368366/34715610-7ce7966c-f52d-11e7-95c9-5862f76a4ada.JPG)

Probably the one on web/bbs. May this be the issue?

**Comment by rvolz** [comment]
The changes you made are correct. The absence of log files means that the error happens before the bbs app is called/initialized. So probably some setup problem.

Version 1.4 consists mostly of bug fixes for 1.3, but includes some minor modifications for Apache 2.3. You could try to install [the archive for 1.3.6](https://github.com/rvolz/BicBucStriim/releases/tag/v1.3.6) instead.

**Comment by DoctorDri** [comment]
I downloaded, uploaded and expanded the zip file for 1.3.6

Checked permissions for bbs2 folder, all like the original bbs. I even allowed Read & Write for Everyone for a while to test... to no avail, same error:

![capture3](https://user-images.githubusercontent.com/21368366/34716430-20020222-f530-11e7-8e8a-31729741b2d4.JPG)

Meanwhile, BBS is working perfectly on web/bbs,

**Comment by DoctorDri** [comment]
I tried calling bbs2 internally using my local IP by I got the same result.

I also deleted the bbs2 folder and inserted the index.php described, exactly same error.

I'm out of ideas, since the port forwarding of the router seems to be ok because the original bbs folder works perfectly.

**Comment by rvolz** [comment]
Tested on my Synology with PHP 5.6 and Apache 2.2:

1. Copy the bbs folder : `cp -R bbs bbs2`
2. Try to call http://myserver/bbs2:  error, exception (database)
3. Change the owner of folder bbs2 to http user (apache) : `chmod -R http:http bbs2`
4. Try to call http://myserver/bbs2:  works  

If that doesn't work for you, you could enable logging for the Apache server and then analyze the error logs of the web server to find what causes this behavior.

**Comment by DoctorDri** [comment]
I finally had time to look into this and after several attempts and a lot of reading, **I managed to solve the issue**. Now I have multiple instances of BicBucStriim 1.4 on my Synology 918+ working well.

Here's what I did:

1.  I didn't know how to SSH into my server, so I couldn't run the commands above. I configured my server to allow SSH and tried to access it with WinSCP, but I couldn't do anything cause it needs root permissions. It wasn't easy because of how DSM 6  works, but finally a combination of the programs PuTTY and WinSCP allowed me to create folders and assign permissions as root.

2. To my surprise, the folders "/volume1/web" and "/var/services/web" are not the same folder. So all the "bbs" folders and attempts that I was trying on "/volume1/web" didn't really do anything. The real deal is "/var/services/web"

3. On "/var/services/web" I created a new folder "bbs8" owned by "http" and with permissions to read and write. Copied the unpacked files for BBS 1.4 there and made sure that permissions where recursive for it.

4. It worked! Now accessing http://myserver/bbs8 is possible and BBS works fine both in my Spanish and my English libraries at the same time.


Thanks for your help. BicBucStriim is a program that modern NAS need.

One last question: how can we donate money to the project?
